vdev/src/commands/test.rs

```diff
@@ -0,0 +1,56 @@
+use anyhow::Result;
+use clap::Args;
+use std::collections::BTreeMap;
+
+use crate::app;
+use crate::platform;
+use crate::testing::{config::RustToolchainConfig, runner::*};
+
+/// Execute tests
+#[derive(Args, Debug)]
+#[command()]
+pub struct Cli {
+    /// Extra test command arguments
+    args: Option<Vec<String>>,
+
+    /// Whether to run tests in a container
+    #[arg(short = 'C', long)]
+    container: bool,
+
+    /// Environment variables in the form KEY[=VALUE]
+    #[arg(short, long)]
+    env: Option<Vec<String>>,
+}
+
+impl Cli {
+    pub fn exec(&self) -> Result<()> {
+        let toolchain_config = RustToolchainConfig::parse(app::path())?;
+        let runner = get_agent_test_runner(self.container, toolchain_config.channel);
+
+        let mut env_vars = BTreeMap::new();
+        if let Some(extra_env_vars) = &self.env {
+            for entry in extra_env_vars {
+                if let Some((key, value)) = entry.split_once("=") {
+                    env_vars.insert(key.to_string(), value.to_string());
+                } else {
+                    env_vars.insert(entry.to_string(), "".to_string());
+                }
+            }
+        }
+
+        let mut args = vec!["--workspace".to_string()];
+        if let Some(extra_args) = &self.args {
+            args.extend(extra_args.to_owned());
+
+            if !(self.container || extra_args.contains(&"--features".to_string())) {
```

Comment:
`"blah"..to_string()` is basically just `"blah"`, so this should work:

```suggestion
            if !(self.container || extra_args.contains("--features")) {
```

Reply:
![s](https://user-images.githubusercontent.com/9677399/202209400-4cdd672b-e97f-466c-b119-b19ea21d97b1.PNG)